### PR TITLE
Retry network calls during initialization

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Certain network issues cannot be handled automatically by the library consumer, so it does not make sense to throw these errors. Instead, grammY will now print some debug output and retry.

### CODE REVIEW

**Comment by darvesh** [comment]
Is retry with no backoff a good idea though? Especially when it returns 429

**Comment by KnorpelSenf** [comment]
Good point, we should at least wait the recommended number of seconds for 429 errors.

I agree that it's a little weird to wait 3 seconds during polling, but not during initialisation. However, I'd rather go with removing the delay for polling rather than introducing one here. If retrying is the recovery strategy from flaky network issues, I don't see how trying a few more times could cause any harm.

**Review by KnightNiwrem — APPROVED** [review]
Any plans to bound the maximum number of retries?

**Comment by KnorpelSenf** [comment]
What would happen after the last retry? Crashing? The point is that we acknowledge these kinds of errors to be temporary and unpredictable. We cannot recover from them or do anything about it. Retrying reliably and infinitely rather than stopping at an arbitrary point is the logical step here.

This PR builds upon advice from the TDLib creator himself. The conversation starts at https://t.me/tdlibchat/22922

File: src/bot.ts [hunk 1]
```diff
@@ -448,3 +450,29 @@ you can circumvent this protection against memory leaks.`);
         this.errorHandler = errorHandler;
     }
 }
+
+async function withRetries(task: () => Promise<unknown>) {
+    let success = false;
+    while (!success) {
+        try {
+            await task();
+            success = true;
```
Comment (dzek69) [inline src/bot.ts:459]: i'd return the result of `task()` here, so the function is more useful for the future :P

Comment (KnorpelSenf) [reply src/bot.ts:459]: Good point

Comment (KnorpelSenf) [reply src/bot.ts:459]: Good point!

Comment (KnorpelSenf) [reply src/bot.ts:459]: This will be done in #182.

**Pull request merged**